src/aspire/reconstruction/mean.py

```diff
@@ -200,12 +200,42 @@ def conj_grad(self, b_coef, tol=1e-5, regularizer=0):
         tol = tol or config.mean.cg_tol
         target_residual = tol * norm(b_coef)
 
+        # callback setup
+        self.i = 0  # iteration counter
+
         def cb(xk):
+            self.i += 1  # increment iteration count
+
             logger.info(
-                f"Delta {norm(b_coef - self.apply_kernel(xk))} (target {target_residual})"
+                f"[Iter {self.i}]: Delta {norm(b_coef - self.apply_kernel(xk))} (target {target_residual})"
             )
 
-        x, info = cg(operator, b_coef.flatten(), M=M, callback=cb, tol=tol, atol=0)
+            # Do checkpoint at `checkpoint_iterations`,
+            _do_checkpoint = (
+                self.checkpoint_iterations
+                and (self.i % self.checkpoint_iterations) == 0
+            )
+            # or the last iteration when `maxiter` provided.
+            if self.maxiter:
+                _do_checkpoint |= self.i == (self.maxiter - 1)
+
+            # Optional checkpoint
+            if _do_checkpoint:
+                # Construct checkpoint path
+                path = f"{self.checkpoint_prefix}_iter{self.i}.npy"
```

Comment:
Can we zero-prefix this (say `{self.i:04d}`) to make sorting easier?

Reply:
sure, added.